### ISSUE TITLE
Add `agent.activation_method` field

### ISSUE THREAD

**Issue by AlexanderWert** [elastic/apm-server#9622]
To capture the way an APM agent is installed we need a new **metadata** field `agent.activation_method` in the Intake v2. The field should be added to trace event documents (transaction, span, error, metrics) in the final Elasticsearch documents. APM Agents will use this field to provide information on how the were installed / used (e.g. auto-attachment vs. manual attachment, etc.). 

The values of this field will be agent-specific. An agent spec will follow to describe the values.

**Comment by AlexanderWert** [comment]
@carsonip Please let us know when you start working on this, so that we finally align on the name for the new field!

**Comment by gregkalapos** [comment]
The naming was discussed a lot in the agents team and the term "installation" doesn't fit well all the agents.

We ended up calling this "agent activation methods". That's also what we have in the spec: https://github.com/elastic/apm/pull/732

We should be aligned on this in the intake API, so I propose let's call it `agent.activation_method` - in any case, let's prefer the term "activation" over "installation".  - I also renamed it in the title.
